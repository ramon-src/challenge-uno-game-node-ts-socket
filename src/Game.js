import MakeCards from "./MakeCards";
import MakeDeck from "./MakeDeck";

function Game() {
  this.sessionId = "";
  this.usersOrder = []; // array containing ids
  this.users = [];
  this.turn = null; // turn number
  this.discardPile = [];
  this.deck = null; // deck object

  this.giveCards = () => {
    users.forEach((user) => {
      user.handsCards();
    });
  };

  this.initialize = () => {
    const cards = new MakeCards().build();
    this.deck = new MakeDeck(cards).shuffle();
  };
}

export default Game;
