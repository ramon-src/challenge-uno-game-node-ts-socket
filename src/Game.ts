import { Deck as IDeck, Player as IPlayer } from "./Interfaces";
import Cards from "./MakeCards";
import Deck from "./Deck";
import Player from "./Player";

function Game() {
  this.sessionId = "";
  this.playersOrder = []; // array containing ids
  this.players = [] as IPlayer[];
  this.turn = null; // turn number
  this.discardPile = [];
  this.deck = null; // deck object

  this.addPlayer = (id: string) => {
    this.players.push(new Player(id));
    this.playersOrder.push(id);
  };

  this.giveCards = (deck: IDeck) => {
    const playersInGame = this.playersOrder.length;
    const limitOfCardsToGive = playersInGame * 7;
    let playersIterator = 0;

    const addCardToPlayer = () => {
      this.players[playersIterator].buyCard(deck);

      if (playersIterator < playersInGame - 1) {
        playersIterator += 1;
      } else {
        playersIterator = 0;
      }
    };

    for (let i = 1; i <= limitOfCardsToGive; i++) {
      addCardToPlayer();
    }
  };

  this.initialize = () => {
    const cards = new Cards().build();
    this.deck = new Deck(cards).shuffle();

    this.giveCards(this.deck);
  };
}

export default Game;
