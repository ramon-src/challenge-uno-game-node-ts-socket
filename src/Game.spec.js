import Game from "./Game";

describe("Game", () => {
  it("should start the game with default values", () => {
    const game = new Game();

    expect(game.deck).toBeNull();
  });

  xit("should initialize the game", () => {
    const game = new Game();

    game.initialize();
  });
});
