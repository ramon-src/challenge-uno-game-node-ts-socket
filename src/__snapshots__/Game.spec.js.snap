// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should initialize the game 1`] = `null`;
